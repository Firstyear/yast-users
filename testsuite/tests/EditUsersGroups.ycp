/**
 * File:	EditUsersGroups.ycp
 * Module:	Users configurator
 * Summary:	Testing adding and removing user from groups (see bug #136267)
 * Author:	Jiri Suchomel <jsuchome@suse.cz>
 * $Id$
 */

{
    // testedfiles: Users.pm

    import "Users";
    import "Mode";
    import "Testsuite";

    map READ = $[
	"passwd": $[
	    "local": $[
		"users": $[
		    "hh": $[
			"uid"		: "hh",
			"uidnumber"	: 500,
			"type"		: "local",
			"userpassword"	: "password",
			"gidnumber":	100,
			"grouplist"	: $[
			    "ggl"	: 1
			],
		    ],
		    "by_uidnumber": $[
			500		: $[ "hh": 1],
		    ],
		    "last_uid": 500,
		    "homes":	$[
		    ],
		    "usernames": $[
			"hh":		1,
		    ],
		    "uids": $[
			500:		1,
		    ]
		],
		"shadow": $[
		],
		"groups": $[
		    "users": $[
			"cn"	:	"users",
			"gidnumber":	100,
			"type"	:	"local",
			"more_users"	: $[
			    "hh":	1
			],
		    ],
		    "ggl": $[
			"cn"	:	"ggl",
			"gidnumber":	1000,
			"type"	:	"local",
			"userlist":	$[
			    "hh":	1
			]
		    ],
		    "by_gidnumber": $[
			100		: $["users":1],
			1000		: $["ggl":1],
		    ],
		    "gids": $[
			100		: 1,
			1000		: 1,
		    ],
		    "groupnames": $[
			"users"		: 1,
			"ggl"		: 1,
		    ]
		],
	    ],
	    "system": $[
		"users": $[
		    "by_uidnumber"	: $[],
		    "last_uid"		: 0,
		    "homes"		: $[],
		    "usernames"		: $[],
		    "uids"		: $[],
		],
		"shadow": $[
		],
		"groups": $[
		    "ggs": $[
			"cn"		: "ggs",
			"gidnumber"	: 10,
			"type"		: "system"
		    ],
		    "by_gidnumber": $[
			10		: $["ggs":1],
		    ],
		    "gids": $[
			10		: 1
		    ],
		    "groupnames": $[
			"ggs"		: 1
		    ]
		],
	    ],
	],
	"etc" : $[
	    "fstab": [],
	    "cryptotab": [],
	    "default": $[
		"useradd": $[
		    "home":	"/home",
		    "groups":	"audio,video",
		    "group":	100
		]
	    ],
	],
	"target": $[
	    "stat"	: $[],
	    "size"	: -1,
	    "tmpdir"	: "/tmp/YaST"
	],
	"product": $[
		"features": $[
		    "USE_DESKTOP_SCHEDULER"	: "no",
		    "IO_SCHEDULER"		: "",
		    "ENABLE_AUTOLOGIN"		: "false",
		    "UI_MODE"			: "simple",
		    "EVMS_CONFIG"		: "no",
		    "INCOMPLETE_TRANSLATION_TRESHOLD"	: "99",
		]
	],
    ];
    map WRITE = $[];
    map EXEC = $[
	"passwd" : $[
	    "init"	: true
	],
	"target" : $[
	    "bash"		: 0,
	    "bash_output"	: $[],
	],
    ];

    Mode::SetTest ("test");

    Testsuite::Test (``(Users::Read ()), [READ, WRITE, EXEC], 0);

    // for home directory checks
    READ["target","stat","isdir"]	= true;

    Users::SelectUserByName ("hh");

    map<string,any> changes = $[
	"grouplist"	: $[
	    "ggs"	: 1,
	]
    ];

    Testsuite::Test (``(Users::EditUser (changes)), [READ, WRITE, EXEC], 0);

    EXEC ["target", "bash_output", "stdout"] = "hh";

    string error = (string) Testsuite::Test(``(Users::CheckUser ($[])), [READ,WRITE,EXEC], 0);

    Testsuite::Dump (sformat ("---- unchanged group 'ggl':\n %1", Users::GetGroupByName ("ggl", "")));

    Testsuite::Dump (sformat ("---- unchanged group 'ggs':\n %1", Users::GetGroupByName ("ggs", "")));

    Testsuite::Test (``(Users::CommitUser ()), [READ, WRITE, EXEC], 0);

    Testsuite::Dump (sformat ("---- changed group 'ggl':\n %1", Users::GetGroupByName ("ggl", "")));

    Testsuite::Dump (sformat ("---- changed group 'ggs':\n %1", Users::GetGroupByName ("ggs", "")));
}
