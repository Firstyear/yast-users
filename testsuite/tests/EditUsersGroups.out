Read	.etc.default.useradd."expire" 0
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Execute	.passwd.init $["base_directory":"/etc", "max_system_gid":"499", "max_system_uid":"499"] true
Read	.passwd.local.users $["by_uidnumber":$[500:$["hh":1]], "hh":$["gidnumber":100, "grouplist":$["ggl":1], "type":"local", "uid":"hh", "uidnumber":500, "userpassword":"password"], "homes":$[], "last_uid":500, "uids":$[500:1], "usernames":$["hh":1]]
Read	.passwd.local.users.by_uidnumber $[500:$["hh":1]]
Read	.passwd.local.shadow $[]
Read	.passwd.local.groups $["by_gidnumber":$[100:$["users":1], 1000:$["ggl":1]], "ggl":$["cn":"ggl", "gidnumber":1000, "type":"local", "userlist":$["hh":1]], "gids":$[100:1, 1000:1], "groupnames":$["ggl":1, "users":1], "users":$["cn":"users", "gidnumber":100, "more_users":$["hh":1], "type":"local"]]
Read	.passwd.local.groups.by_gidnumber $[100:$["users":1], 1000:$["ggl":1]]
Read	.passwd.system.users $["by_uidnumber":$[], "homes":$[], "last_uid":0, "uids":$[], "usernames":$[]]
Read	.passwd.system.users.by_uidnumber $[]
Read	.passwd.system.shadow $[]
Read	.passwd.system.groups $["by_gidnumber":$[10:$["ggs":1]], "ggs":$["cn":"ggs", "gidnumber":10, "type":"system"], "gids":$[10:1], "groupnames":$["ggs":1]]
Read	.passwd.system.groups.by_gidnumber $[10:$["ggs":1]]
Read	.passwd.passwd.pluslines 0
Read	.passwd.shadow.pluslines 0
Read	.passwd.group.pluslines 0
Read	.passwd.local.users.last_uid 500
Read	.passwd.local.users.homes $[]
Read	.passwd.local.users.usernames $["hh":1]
Read	.passwd.local.users.uids $[500:1]
Read	.passwd.system.users.last_uid 0
Read	.passwd.system.users.homes $[]
Read	.passwd.system.users.usernames $[]
Read	.passwd.system.users.uids $[]
Read	.passwd.local.groups.gids $[100:1, 1000:1]
Read	.passwd.local.groups.groupnames $["ggl":1, "users":1]
Read	.passwd.system.groups.gids $[10:1]
Read	.passwd.system.groups.groupnames $["ggs":1]
Return	
Return	
Execute	.target.bash_output "echo 'hh' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"hh"]
Return	
Dump	---- unchanged group 'ggl':
Dump	 $["cn":"ggl", "gidnumber":"1000", "type":"local", "userlist":$["hh":"1"]]
Dump	---- unchanged group 'ggs':
Dump	 $["cn":"ggs", "gidnumber":"10", "type":"system"]
Return	true
Dump	---- changed group 'ggl':
Dump	 $["cn":"ggl", "gidnumber":"1000", "modified":"edited", "org_cn":"ggl", "org_gidnumber":"1000", "type":"local", "userlist":$[], "what":"user_change"]
Dump	---- changed group 'ggs':
Dump	 $["cn":"ggs", "gidnumber":"10", "modified":"edited", "org_cn":"ggs", "org_gidnumber":"10", "type":"system", "userlist":$["hh":"1"], "what":"user_change"]
